libsplinter/src/error/invalid_state.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Module containing InvalidStateError implementation.
+
+use std::error;
+use std::fmt;
+
+/// An error returned when an operation cannot be completed because the state of the struct is
```

Comment:
I don't know if it's always the case that the thing that returns this error is a struct (could be a trait method, for instance).

Reply:
Yeah, I wanted to say 'object' here, if it were another language. Traits are implemented on structs and it would be the data in the struct that would make it invalid. The only thing not backed by a struct would be a standalone function, but then its probably an InvalidArgumentError.

Reply:
I changed this to say "underlying struct" to attempt to make it more clear.